packages/enhanced/src/lib/sharing/ConsumeSharedPlugin.ts

```diff
@@ -259,15 +258,17 @@ class ConsumeSharedPlugin {
                     data,
                     packageName,
                   );
-                  if (typeof requiredVersion !== 'string') {
```

Comment:
we should still keep the required version warning if the recursive lookup doesnt return the package for whatever reason.

What happens if someone has a package.json outside their app project, like in the root of their FS? This will crawl upwards across the whole fs?

Reply:
recursive lookup will stop under the following condition: https://github.com/module-federation/core/blob/ea53f407e9df0c6018904adc22af90437a384c2b/packages/enhanced/src/lib/sharing/utils.ts#L355  
and this warning will be registered
https://github.com/module-federation/core/blob/ea53f407e9df0c6018904adc22af90437a384c2b/packages/enhanced/src/lib/sharing/ConsumeSharedPlugin.ts#L248  
  
but we can actually reach the root of fs. Maybe we should stop searching at some upwards?

Reply:
yeah, maybe try with compiler.context as the edge boundary? Im not sure if that will work in your use case, but it probbably will be a good idea to somehow prevent it from searching the entire fs all the way up to root if it doesnt find something. Like keep it from leaving the current repo, usually i use compiler.context to find the boundary of the compiler.

Reply:
Yeah. Okay well I think we can run with this. I'll review and merge. Then will port to rust